streaming/src/main/scala/io/gearpump/streaming/task/Subscription.scala

```diff
@@ -182,7 +184,28 @@ class Subscription(
   }
 
   def minClock: TimeStamp = {
-    minClockValue.min
+    if (minClockValue.isEmpty) {
```

Comment:
It doesn't look right to do this in minClock
as minClock is supposed to be function with no side effect.